docs/index.md

```diff
@@ -1,91 +1,121 @@
-            result = this.Buffer[this.TakeAt++];
-            Monitor.Pulse(this.SyncObject);
-        }
-
-        return result;
+    if (line != null) {
+      caption.style.display="block";
+      caption.innerHTML = line[1]
+    } else {
+      caption.style.display="none";
     }
-}
-```
-
-This is a producer consumer queue where the `Take` method blocks if the buffer is empty and the
-`Put` method blocks if the buffer has reached it's maximum allowed capacity and while the code
-looks correct it contains a nasty deadlock bug.
-
-Writing concurrent code is tricky and with the popular `async/await` feature in the C# Programming
-Language it is now extremely easy to write concurrent code.
-
-But how do we test concurrency in our code? Testing tools and frameworks have not kept up with the
-pace of language innovation and so this is where `Coyote` comes to the rescue. When you use Coyote
-to test your programs you will find concurrency bugs that are very hard to find any other way.
-
-Coyote rewrites your `async tasks` in a way that allows the [coyote test](tools/testing.md) tool to
-control all the concurrency and locking in your program and this allows it to find bugs using
-intelligent [systematic testing](core/systematic-testing.md).
-
-For example, if you take the `BoundedBuffer.dll` from the above sample you can do the following:
-
-```
-coyote rewrite BoundedBuffer.dll
-coyote test BoundedBuffer.dll -m TestBoundedBufferMinimalDeadlock --iterations 100
-```
+  }
+
+  $(document).ready(function () {
+      video  = $("#shortintro")[0];
+      caption = $("#caption")[0];
+      caption.style.display="none";
+      video.ontimeupdate = function() { show_captions(video, caption); };
+  });
+
+</script>
+
+Coyote helps write powerful, expressive tests for your code. You can declare sources of
+non-determinism (such as timers, failures, etc.) as part of your tests. The Coyote testing tool can
+_systematically_ explore a large number of interleavings of concurrent operations as well as
+non-deterministic choices so that it covers a large set of behaviors in a very short time. This is
+different from _stress testing_. Coyote takes control of the concurrency so that it can manipulate
+every possible scheduling. With appropriate _mocking_, Coyote can also do this in "developer" mode
+on a single laptop with little or no dependence on the bigger production environment.
+
+Coyote is not a verification system. It does not use theorem proving to make correctness guarantees,
+instead it uses intelligent search strategies to drive systematic testing, based on deep
+understanding of concurrency primitives that you have used in your code. This approach has proven to
+work well for large production teams, including many teams in Microsoft Azure because it has a small
+barrier to entry with almost immediate benefits for those who adopt it.
+
+Coyote does not require that a team starts from scratch and rebuild their system. Often it is too
```

Comment:
Maybe we can say after the first sentence, something like the following?

> Coyote does not require that a team starts from scratch and rebuild their system. In fact, for task-based programs the code does not need to change at all, as Coyote uses binary rewriting to intercept sources of task concurrency.

Reply:
Pushing to bigger TODO work item on this topic.

Reply:
Btw, my thinking was to reinforce the point that its not so hard to be used for typical task scenarios (and the users read it early on) :)